File: src/main/java/my/study/repository/design/pattern/decorator/java/UseDemo.java
package my.study.repository.design.pattern.decorator.java;

/**
 * @author djh on  2018/10/2 20:49
 * @E-Mail 1544579459@qq.com
 */
public class UseDemo {
    public static void main(String[] args) {
        ConcreteComponent concreteComponent = new ConcreteComponent();
        BaseDecorate decorateA = new DecorateA(concreteComponent);
        decorateA.operator();
    }
}

File: src/main/java/my/study/repository/design/pattern/decorator/kotlin/UseDemo.kt
package my.study.repository.design.pattern.decorator.kotlin

/**
 * @author djh on  2018/10/2 20:48
 * @E-Mail 1544579459@qq.com
 */
object UseDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        val decorateA = DecorateA()
        decorateA.operator()
    }
}